API/fleece/slice.hh

```diff
@@ -287,6 +287,7 @@ namespace fleece {
         // FLSlice interoperability:
         constexpr slice(const FLSlice &s) noexcept STEPOVER           :pure_slice(s.buf,s.size) { }
         inline explicit operator FLSliceResult () const noexcept;
+        explicit slice(const FLSliceResult &sr)                     :pure_slice(sr.buf, sr.size) { }
```

Comment:
Why does this need to be different from "constexpr slice(const FLSlice &s) noexcept STEPOVER" ?

Reply:
It doesn't make sense to be `constexpr` because `FLSliceResult` can't exist at compile-time (no heap.)

But it should be STEPOVER (which just keeps debugger Step-In commands from stepping into it.)

Reply:
* It doesn't need to be constexpr because you can't have FLSliceResult instances at compile-time (no heap!)
* I made it `explicit` because explicit conversions can be dangerous and do unintended things. It's an uncommon conversion, and making it explicit didn't break anything in LiteCore.
* I should probably make it STEPOVER; all that does is stop the debugger from trying to step into it, which is annoying.